File: src/app/Models/Entities/Catalog.ts

import { Model } from 'app/Models/Model';

export interface Catalog extends Model {

	/**
	 * 显示名称
	 */
    name: string;

	/**
	 * 分类值
	 */
    value: string;

	/**
	 * 分类的类别
	 */
    type: string;

	/**
	 * 是否为顶级分类
	 */
    isTop: number;

	/**
	 * 上级/顶级(如果存在)分类
	 */
    topCatalog: Catalog;

}

File: WebApp/DB/Entity/CataLog.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace WebApp.DB
{
    [JsonObject(IsReference = true)]
    public partial class Catalog
    {
        public Catalog()
        {
            InverseTopCatalog = new HashSet<Catalog>();
            Resource = new HashSet<Resource>();
        }

        public Guid Id { get; set; }
        public DateTime CreatedTime { get; set; }
        public int IsTop { get; set; }
        public string Name { get; set; }
        public int? Status { get; set; }
        public Guid? TopCatalogId { get; set; }
        public string Type { get; set; }
        public DateTime UpdatedTime { get; set; }
        public string Value { get; set; }

        public Catalog TopCatalog { get; set; }

        [JsonIgnore]
        public ICollection<Catalog> InverseTopCatalog { get; set; }
        public ICollection<Resource> Resource { get; set; }
    }
}
